module/VuFind/src/VuFind/Recommend/LibGuidesAZResultsDeferred.php

```diff
@@ -0,0 +1,136 @@
+ * @package  Recommendations
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @author   Maccabee Levine <msl321@lehigh.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:recommendation_modules Wiki
+ */
+class LibGuidesAZResultsDeferred implements RecommendInterface
+{
+    /**
+     * Raw configuration parameters
+     *
+     * @var string
+     */
+    protected $rawParams;
+
+    /**
+     * Current search query
+     *
+     * @var string
+     */
+    protected $lookfor;
+
+    /**
+     * Configuration parameters processed for submission via AJAX
+     *
+     * @var string
+     */
+    protected $processedParams;
+
+    /**
+     * Store the configuration of the recommendation module.
+     *
+     * @param string $settings Settings from searches.ini.
+     *
+     * @return void
+     */
+    public function setConfig($settings)
+    {
+        $this->rawParams = $settings;
+    }
+
+    /**
+     * Called before the Search Results object performs its main search
+     * (specifically, in response to \VuFind\Search\SearchRunner::EVENT_CONFIGURED).
+     * This method is responsible for setting search parameters needed by the
+     * recommendation module and for reading any existing search parameters that may
+     * be needed.
+     *
+     * @param \VuFind\Search\Base\Params $params  Search parameter object
+     * @param \Laminas\Stdlib\Parameters $request Parameter object representing user
+     * request.
+     *
+     * @return void
+     */
+    public function init($params, $request)
+    {
+        // Parse out parameters:
+        $settings = explode(':', $this->rawParams);
+
+        // Make sure all elements of the params array are filled in, even if just
+        // with a blank string, so we can rebuild the parameters to pass through
+        // AJAX later on!
+        for ($i = 0; $i < 2; $i++) {
+            $settings[$i] ??= '';
+        }
+
+        // Map the user-specified query field to 'lookfor' for simplicity:
+        $this->lookfor
+            = $request->get(empty($settings[0]) ? 'lookfor' : $settings[0], '');
+        $settings[0] = 'lookfor';
+        $this->processedParams = implode(':', $settings);
+    }
+
+    /**
+     * Called after the Search Results object has performed its main search.  This
+     * may be used to extract necessary information from the Search Results object
+     * or to perform completely unrelated processing.
+     *
+     * @param \VuFind\Search\Base\Results $results Search results object
+     *
+     * @return void
+     */
+    public function process($results)
+    {
+        // No action needed
+    }
+
+    /**
+     * Get the URL parameters needed to make the AJAX recommendation request.
+     *
+     * @return string
+     */
+    public function getUrlParams()
+    {
+        return 'mod=LibGuidesAZResults&params=' . urlencode($this->processedParams)
```

Comment:
This file is a complete copy/paste from EDSResultsDeferred, except for the 'mod' param in this one line.  I think separately it should be refactored into some sort of SearchObjectDeferred.php?  Saying this partially because once this PR is through, I plan to create additional recommendation modules for LibGuides (traditional, not AZ) and EPF, which will copy/paste this twice more.

Reply:
Agreed, it probably makes sense to create an abstract base class, and then make all of these other versions subclasses that just override a property or two.

Reply:
Will do next.